Arelle/ixbrl-viewer
# Calculation 1.1 support and calculation validation

#### Reason for change

Calculations 1.1 makes calculation checking much more useful, but users need tools to visualise and understand checks.  This PR adds the ability to validate calculations in the viewer, using both Calc 1.0 and 1.1 semantics, and adds a calculation inspector to visualise calculations.

#### Description of change

* New checkbox to enable Calculations 1.1 mode

* Calculation validation status is now shown in the inspector:

![image](https://github.com/Arelle/ixbrl-viewer/assets/45077928/8b71c023-f16e-4e37-804e-5372ae2bf909)

The calculation applied depends on the calculation mode

* The "details" link provides a calculation inspector:

![image](https://github.com/Arelle/ixbrl-viewer/assets/45077928/7d4a1912-207b-4ad5-8aa2-1dc123a44d57)
 
The PR also adds a `copyScript` parameter to `generateViewer` to disable the copying of the JS script to the same directory as the viewer, as this makes development more awkward (default is as before).

#### Steps to Test

Open any report with calculations.  Check the inspector in both calc 1.0 and calc 1.1 modes.  They should generally yield the same results, although the inspectors should be different.

The calculations will be behave differently if the calculations are not exact, but are consistent to within their reported accuracy.  They will also be different in the face of non-complete duplicates.

**review**:
@Arelle/arelle
@paulwarren-wk
